src/main/java/seedu/address/logic/commands/DeletePersonCommand.java

```diff
@@ -0,0 +1,128 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * When no event is selected, deletes a person identified using it's displayed index from the address book.
+ * When an event is selected, deletes a person identified using it's displayed index from the event.
+ */
+public class DeletePersonCommand extends Command {
+
+    public static final String COMMAND_WORD = "delp";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": When no event is selected, deletes the person identified by the index number used in the displayed "
+            + "person list; when an event is selected, deletes a person identified using it's displayed index from "
+            + "the event.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_PERSON_GLOBAL_SUCCESS = "Deleted Person: %1$s from the address book.";
+    public static final String MESSAGE_DELETE_PERSON_EVENT_SUCCESS = "Deleted Person: %1$s from the selected event.";
+    public static final String MESSAGE_NO_SUCH_PERSON = "This person is not found in the selected event.";
+
+
+    private final Index targetIndex;
+
+    /**
+     * Constructor of DeletePersonCommand
+     *
+     * @param targetIndex Index of person user wants to delete.
+     */
+    public DeletePersonCommand(Index targetIndex) {
+        requireNonNull(targetIndex);
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.isAnEventSelected()) {
+            return deleteFromGlobal(model);
+        } else {
+            return deleteFromEvent(model);
+        }
```

Comment:
Make the condition `(model.isAnEventSelected())` instead for readability, then reverse the statements accordingly